dlt/extract/pipe.py

```diff
@@ -697,6 +708,39 @@ def _get_source_item(self) -> ResolvablePipeItem:
         except Exception as ex:
             raise ResourceExtractionError(pipe.name, gen, str(ex), "generator") from ex
 
+
+
+    def _get_source_item_round_robin(self) -> ResolvablePipeItem:
+        try:
+            # get items from last added iterator, this makes the overall Pipe as close to FIFO as possible
+            self._round_robin_index = self._round_robin_index % len(self._sources)
```

Comment:
I'd start the index at -1 and increase it by 1 here then takin modulo in the next line. so we save on assignment. I know I'm pedantic but this is the most frequently called piece of code